runtime/oti/VMHelpers.hpp

```diff
@@ -2125,7 +2143,15 @@ class VM_VMHelpers
 		} while (complementGCConcurrentState != (returnedState & complementGCConcurrentState));
 
 		/* if returned state does not contain carrier ID, return that we won */
-		return (complementGCConcurrentState == returnedState);
+		ret = complementGCConcurrentState == returnedState;
```

Comment:
I hope this whole method can be simplified. The main assumption is that if carrierID is set, we don't even need to try an atomic, since concurrent scan is not necessary, due to pre/post mount actions.

now steps should go like this
- read state
- if carrier id is set exit (no need to scan) 
- if this GC scan flag is set exit (no need to scan)
- 'or' this GC scan flag to the local state (complement GC might or might not be set, so 'or' will not change it)
- try atomic to install our flag
- if won, exit (need to scan)
- if lost, due to other GC changing its state retry from the very start (no need to retry if carrier thread won mounting)